sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PathsHelper.java

```diff
@@ -441,24 +439,57 @@ public static PathInfo parseNameSegments(String resourceUrl, String[] segments)
         if (segments.length % 2 == 0) {
             // even number, assume it is individual resource
             if (isResourceType(segments[segments.length - 2])) {
-                return new PathInfo(false, segments[segments.length - 2],
-                        StringEscapeUtils.unescapeJava(StringUtils.strip(resourceUrl, Paths.ROOT)), true);
+                return new PathInfo(false,
+                                    segments[segments.length - 2],
+                                    unescapeJavaAndTrim(resourceUrl),
+                         true);
             }
         } else {
             // odd number, assume it is feed request
             if (isResourceType(segments[segments.length - 1])) {
-                return new PathInfo(true, segments[segments.length - 1],
-                        StringEscapeUtils.unescapeJava(StringUtils.strip(
-                                resourceUrl.substring(0,
-                                        StringUtils.removeEnd(resourceUrl, Paths.ROOT).lastIndexOf(Paths.ROOT)),
-                                Paths.ROOT)),
-                        true);
+                return new PathInfo(true,
+                                    segments[segments.length - 1],
+                                    unescapeJavaAndTrim(
+                                        resourceUrl.substring(0,
+                                                StringUtils.removeEnd(resourceUrl, Paths.ROOT).lastIndexOf(Paths.ROOT))),
+                         true);
             }
         }
 
         return null;
     }
 
+    private static String unescapeJavaAndTrim(String resourceUrl) {
```

Comment:
please also consider a unit test for `unescapeAndTrim()` implementation when escaping or not escaping case.